multi-cluster-orchestrator/samples/hello_world/1-infrastructure/main.tf

```diff
@@ -0,0 +1,298 @@
+/**
+* Copyright 2025 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+# [START gke_mco_hello_world_1_infrastructure]
+locals {
+  hub_cluster_location       = "us-central1"
+  workload_cluster_locations = ["us-west1", "us-east1", "europe-west4"]
+
+  argocd_version = "7.8.28" # Choose from https://github.com/argoproj/argo-helm/releases
+}
+
+data "google_project" "default" {}
+
+### Cluster Service Accounts
+resource "google_service_account" "clusters" {
+  for_each = toset([
+    "hub",
+    "worker"
+  ])
+
+  project      = data.google_project.default.project_id
+  account_id   = "service-account-${each.key}-cluster"
+  display_name = "Service Account for ${each.key} cluster"
+}
+
+# Cluster Service Account Permissions
+resource "google_project_iam_member" "clusters" {
+  for_each = {
+    for o in distinct(flatten([
+      for sa in google_service_account.clusters :
+      [
+        for role in [
+          "roles/container.defaultNodeServiceAccount",
+          "roles/monitoring.metricWriter",
+          # For image streaming
+          "roles/serviceusage.serviceUsageConsumer"
+        ] :
+        {
+          "email" : sa.email,
+          "role" : role,
+        }
+      ]
+    ])) :
+    "${o.email}/${o.role}" => o
+  }
+
+  project = data.google_project.default.project_id
+  role    = each.value.role
+  member  = "serviceAccount:${each.value.email}"
+}
+
+# Workload Identity Permissions
+
+# Service Account of the ArgoCD Application Controller
+# Syncs the applications directly to the workload clusters
+resource "google_project_iam_member" "hub-wi-argo-appsync" {
+  project = data.google_project.default.project_id
+  for_each = toset([
+    "roles/gkehub.viewer",
+    "roles/container.developer",
+    "roles/gkehub.gatewayEditor"
+  ])
+
+  role   = each.value
+  member = "principal://iam.googleapis.com/projects/${data.google_project.default.number}/locations/global/workloadIdentityPools/${data.google_project.default.project_id}.svc.id.goog/subject/ns/argocd/sa/argocd-application-controller"
```

Comment:
note for later; we should add a condition to make it enforce the IdentityProvider to be the Hub Cluster.

Reply:
Also, should the WI for argocd be closer to the argocd helm?

(e.g. IAM code being closer to the things that need it)

Reply:
Currently all the WI are together, but I'll move the ArgoCD WI to the Helm section.

Reply:
We can follow up on the condition in a future release.